src/components/ColumnSettingsList.ts

```diff
@@ -0,0 +1,280 @@
+    if (!args.context.over) {
+      return {
+        ...currentCoordinates,
+        y: currentCoordinates.y - delta,
+      };
+    }
+    return {
+      ...currentCoordinates,
+      y: currentCoordinates.y + delta,
+    };
+  }
+
+  if (event.code === 'Up' || event.code === 'ArrowUp') {
+    // When focus goes above the beginning of the list, funky things happen. Try to move back into the list.
+    if (!args.context.over) {
+      return {
+        ...currentCoordinates,
+        y: currentCoordinates.y + delta,
+      };
+    }
+    return {
+      ...currentCoordinates,
+      y: currentCoordinates.y - delta,
+    };
+  }
+  return undefined;
+};
+
+// Exported for testing purposes only.
+export const dragEndNotifier = ({ active, over, items, onChange }) => {
+  if (over && active.id !== over?.id) {
+    const activeIndex = items.findIndex(({ id }) => id === active.id);
+    const overIndex = items.findIndex(({ id }) => id === over.id);
+
+    onChange(arrayMove(items, activeIndex, overIndex));
+    // This is necessary because of the virtualized list. Without this, focus after a drag ends on up on the drag
+    // handle in the row where the element was dragged from, not on the drag handle in the row it was dropped.
+    // In a setTimeout so that the re-rendering has already happened.
+    setTimeout(function () {
+      const handle = document.getElementById(dragHandleId(active?.id));
+      handle?.focus();
+    });
+  }
+};
+
+export const ColumnSettingsList = ({ items, onChange, toggleVisibility }: ColumnSettingsListProps) => {
+  const sensors = useSensors(
+    useSensor(PointerSensor),
+    useSensor(KeyboardSensor, {
+      coordinateGetter: customCoordinatesGetter,
+      scrollBehavior: 'smooth',
+      keyboardCodes: {
+        start: ['Space', 'Enter'],
+        cancel: ['KeyQ'], // Default is Escape, but that causes the modal to close.
+        end: ['Space', 'Enter'],
+      },
+    })
+  );
+
+  const renderItem = (item, index) => {
+    return h(SortableItem, { id: item.id, name: item.name, index: getPosition(item.id) }, [
+      h(DragHandle),
+      h(Fragment, [
+        h(
+          TooltipTrigger,
+          {
+            // Since attribute names don't contain spaces, word-break: break-all is necessary to
+            // wrap the attribute name instead of truncating it when the tooltip reaches its
+            // max width of 400px.
+            content: span({ style: { wordBreak: 'break-all' } }, [item.name]),
+          },
+          [
+            label(
+              {
+                style: {
+                  lineHeight: `${rowHeight}px`,
+                  ...Style.noWrapEllipsis,
+                },
+              },
+              [
+                h(Checkbox, {
+                  'aria-label': `Show "${item.name}" in table`,
+                  checked: item.visible,
+                  onChange: () => {
+                    if (!_.isUndefined(index)) toggleVisibility(index);
```

Comment:
Brackets around the `if` body would be nice here.

Reply:
Actually, I can remove the conditional. This was left over from when I was rendering a different view (SortableOverlay) during the drag itself that didn't have an index related to it. I ended up ditching that because I found it more confusing than the existing behavior.